traitlets/traitlets.py

```diff
@@ -88,6 +87,59 @@
 class TraitError(Exception):
     pass
 
+
+class EventReport(FrozenBunch):
+    """An object containing callback information for particular events
+
+    Event report classes provide a base class for all FrozenBunch
+    instances being passed to event callbacks. At minimum, a report
+    has a 'type' key/attribute indicating the type of event being
+    reported. Outside of this, it's up to subclasses to add more
+    information.
+    """
+
+    _etype = None
+
+    def __init__(self, *args, **kwargs):
+        super(EventReport, self).__init__(*args, **kwargs)
+        if "type" not in self:
+            dict.__setitem__(self, "type", self._etype)
+
+
+class DefaultEvent(EventReport):
+    """An object holding information about a trait's generated default"""
+
+    _etype = 'default'
+
+    def __init__(self, name=None, value=Undefined, owner=None):
+        super(DefaultEvent, self).__init__(name=name, value=value, owner=owner)
+
+
+class ChangeEvent(EventReport):
+    """An object holding information for trait change callbacks"""
+
+    _etype = 'change'
+
+    def rollback(self):
+        if self.old is not Undefined:
+            self.owner.set_trait(self.name, self.old)
+        else:
+            del self.owner._trait_values[self.name]
+
+    def compress(self, other):
```

Comment:
Adding these methods on event reports is exposing them to observer/validate handlers. Is that what we want? It doesn't seem so. Or are we adding a `change.rollback()` public API to be called in any observer?